pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -305,6 +307,8 @@ public BrokerService(PulsarService pulsar, EventLoopGroup eventLoopGroup) throws
                 .newSingleThreadScheduledExecutor(new DefaultThreadFactory("pulsar-stats-updater"));
         this.authorizationService = new AuthorizationService(
                 pulsar.getConfiguration(), pulsar().getPulsarResources());
+        this.internalExecutorService = new ExecutorProvider(pulsar.getConfiguration().getNumIOThreads(),
```

Comment:
now only transaction pending ack will use it, so when enable transaction we can init it